Azure-Samples/functions-python-azureml-azurefunctions-deeplearning
# Initial content and code migration

## Purpose

The intent of this repository is to communicate the process of training a model using a Python-based Azure Function and the Azure ML Python SDK, as well as, to provide a code sample for doing so. Training a model with the Azure ML Python SDK involves utilizing an Azure Compute option (e.g. an N-Series AML Compute) - the model is not trained within the Azure Function Consumption Plan. Triggers for the Azure Function could be HTTP Requests, an Event Grid or some other trigger.

The motivation behind this process was to provide a way to automate ML model training/retraining in a lightweight, serverless fashion, once new data is provided and potentially labels which are stored Azure Storage Blob containers. This solution is especially attractive for people familiar with Azure Functions.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x] Other... Please describe: First commit of code and documentation;  a migration from another repo.
```

## How to Test
*  Follow the extensive [Readme.md](https://github.com/Azure-Samples/functions-python-azureml-azurefunctions-deeplearning/blob/initial-migration-mh/README.md)
* Test the code
  * Ensure function is running successfully locally and deployed to the cloud.  Check that the ML model appears in the model Blob Storage account.  Check that the training experiment shows a successful completion in the Azure ML Workspace in the Azure Portal.

## What to Check
Verify that the following are valid
* New, completed Experiment in the the Azure ML Workspace
* New model in the specified model Blob Storage Container

## Other Information


Please review and add comments as needed.  Thank you.

**Also, would it be possible to shorten the name of this repo?  I has functions twice in the name.  Thanks.**
